pipeline/build/build_test.go

```diff
@@ -1,206 +1,142 @@
-	folder, back := testlib.Mktmp(t)
-	defer back()
-	writeGoodMain(t, folder)
-	var binary = filepath.Join(folder, buildtarget.Runtime.String(), "testing")
-	var pre = filepath.Join(folder, "pre")
-	var post = filepath.Join(folder, "post")
+func TestRunPipe(t *testing.T) {
 	var config = config.Project{
 		Builds: []config.Build{
 			{
+				Lang:    "fake",
 				Binary:  "testing",
 				Flags:   "-v",
 				Ldflags: "-X main.test=testing",
-				Hooks: config.Hooks{
-					Pre:  "touch " + pre,
-					Post: "touch " + post,
-				},
-				Goos: []string{
-					runtime.GOOS,
-				},
-				Goarch: []string{
-					runtime.GOARCH,
-				},
-			},
-		},
-		Archive: config.Archive{
-			Replacements: map[string]string{
-				"linux":  "linuxx",
-				"darwin": "darwinn",
+				Targets: []string{"whatever"},
 			},
 		},
 	}
 	var ctx = context.New(config)
 	assert.NoError(t, Pipe{}.Run(ctx))
 	assert.Len(t, ctx.Artifacts.List(), 1)
-	assert.True(t, exists(binary), binary)
-	assert.True(t, exists(pre), pre)
-	assert.True(t, exists(post), post)
 }
 
-func TestRunPipeArmBuilds(t *testing.T) {
+func TestRunFullPipe(t *testing.T) {
 	folder, back := testlib.Mktmp(t)
 	defer back()
-	writeGoodMain(t, folder)
-	var binary = filepath.Join(folder, "linuxarm6", "armtesting")
+	var pre = filepath.Join(folder, "pre")
+	var post = filepath.Join(folder, "post")
 	var config = config.Project{
 		Builds: []config.Build{
 			{
-				Binary:  "armtesting",
+				Lang:    "fake",
+				Binary:  "testing",
 				Flags:   "-v",
-				Ldflags: "-X main.test=armtesting",
-				Goos: []string{
-					"linux",
-				},
-				Goarch: []string{
-					"arm",
-					"arm64",
-				},
-				Goarm: []string{
-					"6",
+				Ldflags: "-X main.test=testing",
+				Hooks: config.Hooks{
+					Pre:  "touch " + pre,
+					Post: "touch " + post,
 				},
+				Targets: []string{"whatever"},
 			},
 		},
 	}
 	var ctx = context.New(config)
 	assert.NoError(t, Pipe{}.Run(ctx))
-	assert.Len(t, ctx.Artifacts.List(), 2)
-	assert.True(t, exists(binary), binary)
-}
-
-func TestBuildFailed(t *testing.T) {
-	folder, back := testlib.Mktmp(t)
-	defer back()
-	writeGoodMain(t, folder)
-	var config = config.Project{
-		Builds: []config.Build{
-			{
-				Flags: "-flag-that-dont-exists-to-force-failure",
-				Goos: []string{
-					runtime.GOOS,
-				},
-				Goarch: []string{
-					runtime.GOARCH,
-				},
-			},
-		},
-	}
-	var ctx = context.New(config)
-	assertContainsError(t, Pipe{}.Run(ctx), `flag provided but not defined: -flag-that-dont-exists-to-force-failure`)
-	assert.Empty(t, ctx.Artifacts.List())
-}
-
-func TestRunPipeWithInvalidOS(t *testing.T) {
-	folder, back := testlib.Mktmp(t)
-	defer back()
-	writeGoodMain(t, folder)
-	var config = config.Project{
-		Builds: []config.Build{
-			{
-				Flags: "-v",
-				Goos: []string{
-					"windows",
-				},
-				Goarch: []string{
-					"arm",
-				},
-			},
-		},
-	}
-	assert.NoError(t, Pipe{}.Run(context.New(config)))
+	assert.Len(t, ctx.Artifacts.List(), 1)
```

Comment:
We could also check the artifact here, to see if we actually built the right thing.